test/fuzz/transformation_swap_function_variables_test.cpp

```diff
@@ -185,22 +186,38 @@ TEST(TransformationSwapFunctionVariables, IsApplicable) {
 
   // Successful transformations
   {
+    auto first_instruction = context->get_def_use_mgr()->GetDef(24);
+    auto second_instruction = context->get_def_use_mgr()->GetDef(28);
     // Swap two OpVariable instructions in the same function.
     TransformationSwapFunctionVariables transformation(24, 28);
+
     ASSERT_TRUE(
         transformation.IsApplicable(context.get(), transformation_context));
+
     ApplyAndCheckFreshIds(transformation, context.get(),
                           &transformation_context);
+
+    ASSERT_EQ(first_instruction, context->get_def_use_mgr()->GetDef(24));
+    ASSERT_EQ(second_instruction, context->get_def_use_mgr()->GetDef(28));
```

Comment:
I think these should be swapped. That is, `first_instruction` should be returned for result id 28 and `second_instruction` - for 24. You could additionally check that their `result_id()` is indeed what we expect.

Also, I'd put these checks after the `IsValidAndWellFormed` below.

Reply:
done